source/common/router/header_formatter.cc

```diff
@@ -125,11 +130,12 @@ RequestInfoHeaderFormatter::RequestInfoHeaderFormatter(const std::string& field_
       return RequestInfo::Utility::formatDownstreamAddressNoPort(
           *request_info.downstreamRemoteAddress());
     };
-  } else if (StringUtil::startsWith(field_name.c_str(), "UPSTREAM_METADATA")) {
+  } else if (field_name.find_first_of("UPSTREAM_METADATA") == 0) {
     field_extractor_ =
         parseUpstreamMetadataField(field_name.substr(sizeof("UPSTREAM_METADATA") - 1));
   } else {
-    throw EnvoyException(fmt::format("field '{}' not supported as custom header", field_name));
+    throw EnvoyException(
+        fmt::format("field '{}' not supported as custom header", absl::StrCat(field_name)));
```

Comment:
std::string(field_name).  Actually it'd be nice if fmt::format could be taught somehow about string_view as this will be a common pattern.

Reply:
Filed #2457.